File: Assets/VirtualCity/ProtoDefine/Java/building/ReqRebuildMessage.java
package com.kingston.jforgame.server.game.building.message;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;
import com.kingston.jforgame.server.game.Modules;
import com.kingston.jforgame.server.game.building.BuildingDataPool;
import com.kingston.jforgame.socket.annotation.MessageMeta;
import com.kingston.jforgame.socket.message.Message;

@MessageMeta(module = Modules.BULIDING,cmd = BuildingDataPool.REQ_REBUILD)
public class ReqRebuildMessage extends Message {
    @Protobuf(order = 1)
    private String buildId;
    /**
     * 0:金币
     * 1:钻石
     */
    @Protobuf(order = 3)
    private int costDiamond;

    public String getBuildId() {
        return buildId;
    }

    public void setBuildId(String buildId) {
        this.buildId = buildId;
    }

    public int getCostDiamond() {
        return costDiamond;
    }

    public void setCostDiamond(int costDiamond) {
        this.costDiamond = costDiamond;
    }
}

File: Assets/VirtualCity/ProtoDefine/ReqRebuildMessage.cs
using System.Collections.Generic;
using ProtoBuf;
namespace ProtoDefine {
[ProtoContract]


public class ReqRebuildMessage {
[ProtoMember(1)]
    public string buildId;
    /**
     * 0:金币
     * 1:钻石
     */
[ProtoMember(2)]
    public int costDiamond;

    public string getBuildId() {
        return buildId;
    }

    public void setBuildId(string buildId) {
        this.buildId = buildId;
    }

    public int getCostDiamond() {
        return costDiamond;
    }

    public void setCostDiamond(int costDiamond) {
        this.costDiamond = costDiamond;
    }
}
}